classes/controller/AdminController.php

```diff
@@ -4789,4 +4795,80 @@ protected function isAnonymousAllowed()
     {
         return $this->allowAnonymous;
     }
+
+    /**
+     * Get the buttons provided by hooks and merge them into the already defined page toolbar buttons
+     *
+     * @throws TypeException
+     */
+    protected function mergeExtraToolbarButtons(): void
+    {
+        $toolbarButtonsCollection = new ActionsBarButtonsCollection();
+
+        // Get previously assigned toolbar buttons of the controller
+        $controllerButtons = $this->page_header_toolbar_btn;
+        if (!empty($controllerButtons)) {
+            // Build ActionsBarButton based on array setting from the controller and add it to collection
+            foreach ($controllerButtons as $controllerButton) {
+                $toolbarButtonsCollection->add(
+                    new ActionsBarButton(
+                        $controllerButton['class'] ?? '',
```

Comment:
```suggestion
                        $controllerButtonId,
                        $controllerButton['class'] ?? '',
```

Reply:
As I said in the comment below we need to keep a unique ID, if we use the class as the array key the buttons will be erased when they have the same class